partner_communication/models/communication_job.py

```diff
@@ -390,8 +396,14 @@ def refresh_text(self, refresh_uid=False):
                 fields = self.env['mail.compose.message'].with_context(
                     lang=lang).get_generated_fields(
                     job.email_template_id, [job.id])
+                body = job.report_id.render_qweb_html([job.id])
```

Comment:
Here it looks like you do a second pass of templating, is this really needed? I would probably remove the call of mail.compose.message wizard if that is possible because I have the feeling that the operation is now duplicated which is not optmial for performance.

Reply:
I removed the call to get_generated_fields and as you said it wasn't needed